tests/governance_root_ca_certs.py

```diff
@@ -0,0 +1,110 @@
+
+    primary, _ = network.find_nodes()
+
+    LOG.info("Member builds a root ca cert update proposal with malformed cert")
+    with tempfile.NamedTemporaryFile('w') as f:
+        f.write('foo')
+        try:
+            proposal_body, _ = ccf.proposal_generator.update_root_ca_cert("mycert", f.name)
+        except ValueError:
+            pass
+        else:
+            assert False, "update_root_ca_cert should have raised an error"
+
+    LOG.info("Member makes a root ca cert update proposal with malformed cert")
+    with tempfile.NamedTemporaryFile('w') as f:
+        f.write('foo')
+        proposal_body, _ = ccf.proposal_generator.update_root_ca_cert("mycert", f.name, skip_checks=True)
+        try:
+            proposal = network.consortium.get_any_active_member().propose(primary, proposal_body)
+        except infra.proposal.ProposalNotCreated:
+            pass
+        else:
+            assert False, "Proposal should not have been created"
+
+    LOG.info("Member makes a root ca cert update proposal with valid cert")
+    ca_cert_path = os.path.join(this_dir, 'root_ca_cert.pem')
+    proposal_body, _ = ccf.proposal_generator.update_root_ca_cert("mycert", ca_cert_path)
+    proposal = network.consortium.get_any_active_member().propose(primary, proposal_body)
+    assert proposal.state == ProposalState.Accepted
+    
+    with primary.client(f"member{network.consortium.get_any_active_member().member_id}") as c:
+        r = c.post(
+            "/gov/read", {"table": "ccf.root_ca_cert_ders", "key": "mycert"}
+        )
+        assert r.status == 200, r.status
+        cert_pem_str = open(ca_cert_path).read()
+        cert_ref = x509.load_pem_x509_certificate(cert_pem_str.encode(), crypto_backends.default_backend())
+        cert_kv = x509.load_der_x509_certificate(bytes(r.body), crypto_backends.default_backend())
+        assert cert_ref == cert_kv, f"stored cert not equal to input cert: {cert_ref} != {cert_kv}"
+
+    return network
+
+def run(args):
+    hosts = ["localhost"] * (3 if args.consensus == "pbft" else 2)
+
+    with infra.notification.notification_server(args.notify_server) as notifications:
+        # Lua apps do not support notifications
+        # https://github.com/microsoft/CCF/issues/415
+        notifications_queue = (
+            notifications.get_queue()
+            if (args.package == "liblogging" and args.consensus == "raft")
+            else None
+        )
+
+        with infra.network.network(
+            hosts, args.binary_dir, args.debug_nodes, args.perf_nodes, pdb=args.pdb
+        ) as network:
+            network.start_and_join(args)
+            network = test_cert_store(network, args, notifications_queue)
+
+
+if __name__ == "__main__":
+
+    def add(parser):
+        pass
+
+    # temporary hack...
+    for i, arg in enumerate(sys.argv):
+        if 'gov.lua' in arg:
+            sys.argv[i] = arg.replace('gov.lua', 'gov_root_ca_certs.lua')
+
+    args = infra.e2e_args.cli_args(add=add)
```

Comment:
```suggestion
    args = infra.e2e_args.cli_args(add=add)
    
    # temporary hack...
    args.gov_script = "gov_root_ca_certs.lua"
```

To support this properly, I think we should change the `add_e2e_test` function in CMake; add a `CONSTITUTION` arg, remove `-g ${CCF_DIR}/src/runtime_config/gov.lua` from `CCF_NETWORK_TEST_ARGS`, when constructing `add_test` include either `-g ${PARSED_ARGS}_CONSTITUTION}` or default to `gov.lua`.

Reply:
Should this be a separate PR?